xpk.py

```diff
@@ -4053,6 +4053,41 @@ def get_workload_list(args) -> tuple[int, str]:
     }
   s = ','.join([key + ':' + value for key, value in columns.items()])
 
+  if args.wait_for_job_completion:
+    # Check that the workload exists
+    args.workload = args.wait_for_job_completion
+    workload_exists = check_if_workload_exists(args)
+    if not workload_exists:
+      xpk_print(
+          f'Workload named {args.workload} not found.'
+      )
+      xpk_exit(1)
+
+    # Get the full workload name
+    get_jobset_name_cmd = f'kubectl get workloads | grep jobset-{args.workload}'
+    return_code, return_value = run_command_for_value(
```

Comment:
check the return_code here and exit if failure before running get_jobset_name_cmd step